crates/evm/src/execute.rs

```diff
@@ -499,6 +502,55 @@ where
     }
 }
 
+/// Creates an `EvmState` from a map of balance increments and the current state
+/// to load accounts from.
+/// Zero balance increments are ignored and won't create state entries.
+pub fn balance_increment_state<DB>(
+    balance_increments: &HashMap<Address, u128, DefaultHashBuilder>,
+    state: &mut State<DB>,
+) -> Result<EvmState, BlockExecutionError>
+where
+    DB: Database,
+{
+    let mut create_account =
+        |address: &Address, balance_inc: u128| -> Result<(Address, Account), BlockExecutionError> {
+            let account = state.load_cache_account(*address).map_err(|_| {
+                BlockExecutionError::msg("could not load account for balance increment")
+            })?;
+
+            let transition_account = account
+                .increment_balance(balance_inc)
+                .ok_or(BlockValidationError::IncrementBalanceFailed)?;
+
+            let info = transition_account.info.ok_or_else(|| {
+                BlockExecutionError::msg("empty account info after balance increment")
+            })?;
+
+            let storage = transition_account
+                .storage
+                .into_iter()
+                .map(|(k, v)| {
+                    (
+                        k,
+                        EvmStorageSlot {
+                            original_value: v.original_value(),
+                            present_value: v.present_value,
+                            is_cold: false,
+                        },
+                    )
+                })
+                .collect();
```

Comment:
no storage slots would be modified during this state transition, we can safely skip this

Reply:
this code can be left intact if we change the function to create state update for all post block state transitions (including requests)